src/main/kotlin/com/openlattice/linking/graph/PostgresLinkingQueryService.kt

```diff
@@ -49,16 +49,19 @@ private const val AVG_SCORE_FIELD = "avg_score"
 class PostgresLinkingQueryService(private val hds: HikariDataSource) : LinkingQueryService {
 
 
-    override fun lockClustersForUpdates(
-            clusters: Set<UUID>
-    ): StatementHolder {
+    override fun lockClustersForUpdates(clusters: Set<UUID>): StatementHolder {
         val connection = hds.connection
         connection.autoCommit = false
-        val ps = connection.prepareStatement(LOCK_CLUSTERS_SQL)
-        val arr = PostgresArrays.createUuidArray(connection, clusters)
-        ps.setArray(1, arr)
-        val rs = ps.executeQuery()
-        return StatementHolder(connection, ps, rs)
+
+        val psLocks = connection.prepareStatement(LOCK_CLUSTERS_SQL)
+        clusters.forEach {
+            psLocks.setObject(1, it)
+            psLocks.addBatch()
+        }
+        psLocks.executeBatch()
+        connection.commit()
+
+        return StatementHolder(connection, psLocks, null)
```

Comment:
Just make a statement holder that doesn't take a a result set. Somewhat curious where this statement holder is used as the purpose of a statement holder is so that a postgres iterable can instantiate a postgres iterator on demand.

Reply:
It's used in BackgroundLinkingService, to lock these rows until we insert the new matches. I also should remove the commit() call there.